lib/pbench/server/api/resources/api_key.py

```diff
@@ -63,15 +117,52 @@ def _post(
             raise APIInternalError(str(e)) from e
 
         try:
-            key = APIKey(api_key=new_key, user=user)
+            key = APIKey(api_key=new_key, user=user, name=name)
```

Comment:
The semantics of `DuplicateApiKey` here are interesting. First off, only `id` and `api_key` are constrained to be unique. Which is correct, as we support multiple keys per user, and even if we wanted the `name` to be unique for a particular user (which isn't worth enforcing) they can certainly repeat across users.

But that means that if I `POST /key?name=foo` within whatever time granularity we're talking about for key duplication of a `POST /key?name=bar`, I'm going to get an `OK` for a "duplicate key" even though the single key that stands has a *different* name than what I requested. A `GET` won't return any key with the name "foo" and this is going to be confusing, and I don't like it much.

Maybe we could add a `"name"` into the JWT, but since names aren't guaranteed unique either that only decreases the odds of a duplicate a bit. Another option is to stop pretending it's an innocuous duplication and *fail* with something like `CONFLICT` ... but I really don't like that, either.

Plus ... the addition of `DELETE` complicates matters since the client needs to be really careful only delete if they got a `CREATED` rather than `OK`. It's possible, but not really pleasant.

Another option might be to add some noise into the JWT, like, for example, a sequence number. (Using `APIKey.id` would be great, but that doesn't exist until we `commit` the new `APIKey` and I don't want to guess because that can lead to mis-counts and duplicates; but we could add a `sequence` parameter to the token generator, defaulting to 1 or 0, and, on `DuplicateApiKey` here, we'd increment a local integer and try again overriding the default. This shouldn't be really common since we can only hit a duplicate on two create calls within a clock tick.)

And, either way ... I don't think we have a unit test case that actually *exercises* this path, do we?